Core/GDCore/Events/Event.cpp

```diff
@@ -65,10 +68,40 @@ gd::String BaseEvent::GenerateEventCode(
   return "";
 }
 
+void BaseEvent::PreprocessAsyncActions(const gd::Platform& platform) {
+  if (!CanHaveSubEvents()) return;                        // TODO
+  for (const auto& actionsList : GetAllActionsVectors())  // TODO

```

Comment:
The first one was for me to check if this didn't break anything because I was not sure if that wouldn't break the link event when prototyping. I now know that this shouldn't be the case, and have also tested just in case, that to-do can be removed. The other one is because of how I handle the actions list: I handle them all as having only one actions list, and this would probably break if we add an event with more than one actions list, like an "if-else" event. I do not really know how to improve that though, so I think it'd be best to leave it like that and, if we need to, override the whole function in the event that breaks with this implementation.